File: src/test/java/fixengine/messages/IntegerFieldSpec.java
/*
 * Copyright 2008 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package fixengine.messages;

import jdave.Specification;
import jdave.junit4.JDaveRunner;

import org.junit.runner.RunWith;

/**
 * @author Pekka Enberg 
 */
@RunWith(JDaveRunner.class)
public class IntegerFieldSpec extends Specification<IntegerField> {
    @SuppressWarnings("unchecked") private IntegerField field = new IntegerField(dummy(Tag.class));

    public class AnyField {
        public IntegerField create() {
            return field;
        }

        public void parsesNegativeValues() {
            field.parse("-23");
            specify(field.intValue(), must.equal(-23));
        }

        public void parsesLeadingZeros() {
            field.parse("00023");
            specify(field.intValue(), must.equal(23));
        }

        public void failsToParseFloats() {
            field.parse("1.23");
            specify(field.isFormatValid(), must.equal(false));
        }

        public void failsToParseNonNumericStrings() {
            field.parse("ZZ");
            specify(field.isFormatValid(), must.equal(false));
        }
    }
}

File: core/src/test/scala/stirling/fix/messages/IntegerFieldSpec.scala
/*
 * Copyright 2008 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package stirling.fix.messages

import org.scalatest.WordSpec
import org.scalatest.matchers.MustMatchers
import stirling.fix.tags.fix42.BeginSeqNo

class IntegerFieldSpec extends WordSpec with MustMatchers {
  "IntegerField" should {
    val field = new IntegerField(BeginSeqNo.Tag)
    "parse a negative value" in {
      field.parse("-23")
      field.intValue must equal(-23)
    }
    "parse leading zeroes" in {
      field.parse("00023")
      field.intValue must equal(23)
    }
    "fail to parse a decimal number" in {
      intercept[InvalidValueFormatException] {
        field.parse("1.23")
      }.getMessage must equal("BeginSeqNo(7): Invalid value format: 1.23")
    }
    "fail to parse a non-numeric string" in {
      intercept[InvalidValueFormatException] {
        field.parse("ZZ")
      }.getMessage must equal("BeginSeqNo(7): Invalid value format: ZZ")
    }
  }
}
